libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/SyncDownTarget.java

```diff
@@ -166,6 +167,31 @@ public int cleanGhosts(SyncManager syncManager, String soupName) throws JSONExce
     }
 
     /**
+     * Return ids of non-dirty records (records NOT locally created/updated or deleted)
+     * @param syncManager
+     * @param soupName
+     * @param idField
+     * @return
+     * @throws JSONException
+     */
+    protected SortedSet<String> getNonDirtyRecordIds(SyncManager syncManager, String soupName, String idField) throws JSONException {
```

Comment:
Methods moved down from SyncTarget (parent class) since it's only used by sync down targets.